osutil/stat.go

```diff
@@ -87,3 +87,31 @@ func IsWritable(path string) bool {
 	err := syscall.Access(path, W_OK)
 	return err == nil
 }
+
+// IsDirNotExist tells you whether the given error is due to a directory not existing.
+func IsDirNotExist(err error) bool {
+	switch pe := err.(type) {
+	case nil:
+		return false
+	case *os.PathError:
+		err = pe.Err
+	case *os.LinkError:
+		err = pe.Err
```

Comment:
Nitpick: just fallthrough for all cases?

Reply:
It's a type switch, I don't think that `fallthrough` is allowed here.

Reply:
Yes, clarified with John earlier on IRC, thanks.